### ISSUE TITLE
Virtualbox 'cannot coerce an integer to a string'

### ISSUE THREAD

**Issue by rongcuid** [NixOS/nixpkgs#22487]
## Issue description
When I tried to upgrade today, I get this error:

```
downloading Nix expressions from ‘https://d3g5gsiof5omrk.cloudfront.net/nixos/16.09-small/nixos-16.09.1644.d837e52/nixexprs.tar.xz’...
downloading ‘https://d3g5gsiof5omrk.cloudfront.net/nixos/16.09-small/nixos-16.09.1644.d837e52/nixexprs.tar.xz’... [0/0 KiB, 0.0 KiB/s]
unpacking channels...
building Nix...
building the system configuration...
error: cannot coerce an integer to a string, at /nix/store/mz4vq8b7vrlbwmmlqhdglq475irv7wb7-nixos-16.09.1644.d837e52/nixos/pkgs/applications/virtualization/virtualbox/default.nix:27:13
(use ‘--show-trace’ to show detailed location information)
```

The line corresponding to error is this:

```
    name = "Oracle_VM_VirtualBox_Extension_Pack-${version}-${extpackRev}.vbox-extpack";
```

### Steps to reproduce
Using 16.09-small, installed virtualbox and extension pack. Try to upgrade.

## Technical details

* System: 16.09.1622.e67d70f (Flounder)
* Nix version: nix-env (Nix) 1.11.6
* Nixpkgs version: 16.09.1644.d837e52

**Comment by rongcuid** [comment]
Found a workaround/solution: Changing the line of `extpackRev` to string will make it work. Just not sure whether it is intended, or is it a bug of Nix, so I am not pushing a pull request for now.

**Comment by domenkozar** [comment]
Check git log for that file for hints

**Comment by rongcuid** [comment]
I see lots of CVE's, but I don't have a clue of what I should do.

**Comment by domenkozar** [comment]
fixed in #22542
